### DISCUSSION TITLE
"Or" Operator With Child Rule

### DISCUSSION THREAD

**Discussion by jpiasecki** [microsoft/RulesEngine#329] [Q&A] [answered]
Hi,

I'm a little confused about how the "Or" operator functions with a child rule.  I would have thought that in the following situation, the rule would be considered successful, as long as either the parent or the child rule evaluated successfully:

![image](https://user-images.githubusercontent.com/1817365/157552141-3b1d406f-7347-4622-90d4-7612db56a92a.png)

But it seems like the failure of the child rule results in the entire rule failing, even if I specify an operator of "Or" or "OrElse".  Am I misunderstanding something?

Thanks very much for your time.

**Comment by jpiasecki** [comment, marked as answer]
Playing with this a little further, it would appear that if there are child rules, the expression of the parent rule is not evaluated, and that the parent rule is just a "container" for the child rules.  My apologies.

**Comment by ghost** [comment]
Hi, I actually have the same question. How do you implement rules with precondition or hierarchy? For example, if I got rule says:

Parameter: boatvalue. Expected output: true/false

If boatcolor == "red" then 
--if boatage > 3 then boatvalue > $1000
--else if boatage <=3 and place == "TownA" then boatvalue > $1500
--else if boatage < 1 then  boatvalue > $1800
else if boatcolor == "white" then
--if place == "TownA" then boatvalue > $1500
--else boatvalue > $1700
else 
--if boatage > 5 then boatvalue > $1300
--else boatvalue > $1550

It is a use case in if-else. How do I do this in rulesengine without creating 7 massive rule strings, since parent rule expressions are not in use?
